packages/x-components/src/components/base-variable-column-grid.vue

```diff
@@ -51,11 +51,31 @@
     protected items?: ListItem[];
 
     /**
-     * The columns to render in the grid.
+     * The columns to render by default in the grid. This property is used when the user has not
+     * selected any value in the column picker.
      *
      * @internal
      */
-    protected columns = 0;
+    @Prop({ default: 0 })
+    protected columns!: number;
+
+    /**
+     * The number of columns provided by a user interaction.
+     *
+     * @internal
+     */
+    protected providedColumns = 0;
```

Comment:
We are going to need to provide 0 columns too as is for "automatic" columns number. So no provided columns should be null.

```suggestion
    protected providedColumns: number | null = null;
```